config/config-api/src/main/java/com/thoughtworks/go/domain/config/ConfigurationProperty.java

```diff
@@ -349,4 +348,26 @@ public ConfigurationProperty deserialize(String name, String value, String encry
         }
         return this;
     }
+
+    private void parseSecretParams() {
+        this.secretParamsForValue = SecretParams.parse(getValue());
+    }
+
+    @Override
+    public boolean hasSecretParams() {
+        return !this.secretParamsForValue.isEmpty();
+    }
+
+    @Override
+    public SecretParams getSecretParams() {
+        return this.secretParamsForValue;
+    }
+
+    public String getResolvedValue() {
+        if (hasSecretParams()) {
+            return getSecretParams().substitute(getValue());
+        }
+
+        return getValue();
+    }
```

Comment:
Reason behind introducing a new method: parsing of a secret param and substituting the same needs to check the following cases:
 - not secure values: null nd not blank
 - secure values: null nd successful decryption.

These checks are already performed in `getValue()` method. Hence, to avoid code duplication, decided to introduce a new method. Let me know if this is not ok.